contracts/LifCrowdsale.sol

```diff
@@ -238,13 +231,17 @@ contract LifCrowdsale is Ownable, Pausable {
 
      @param beneficiary Address to which Lif will be sent
      @param weiSent Amount of wei contributed
+     @param rate of tokens per ether at the moment of the contribution
    */
-  function addPrivatePresaleTokens(address beneficiary, uint256 weiSent) onlyOwner {
+  function addPrivatePresaleTokens(
+    address beneficiary, uint256 weiSent, uint256 rate
+  ) onlyOwner {
     require(block.timestamp < startTimestamp);
     require(beneficiary != address(0));
     require(weiSent > 0);
+    require(rate > rate1);
```

Comment:
I'd add a comment explaining this. It'd be something like "validate that rate is higher than TGE rate"

Reply:
:+1: